AppDB/appscale/datastore/scripts/ua_server.py

```diff
@@ -64,6 +65,115 @@
 
 logger = logging.getLogger(__name__)
 
+zk_client = None
+
+table_name = "ua_server"
+
+
+def is_connection_error(err):
+  """ This function is used as retry criteria.
+  It also makes possible lazy load of psycopg2 package.
+
+  Args:
+    err: an instance of Exception.
+  Returns:
+    True if error is related to connection, False otherwise.
+  """
+  return isinstance(err, psycopg2.InterfaceError)
+
+
+retry_pg_connection = retrying.retry(
+    retrying_timeout=10, retry_on_exception=is_connection_error
+)
+
+
+class PostgresConnectionWrapper(object):
+  """ Implements automatic reconnection to Postgresql server. """
+
+  def __init__(self, dsn=None):
+    self._dsn = dsn
+    self._connection = None
+
+  def set_dsn(self, dsn):
+    """ Resets PostgresConnectionWrapper to use new DSN string.
+    Args:
+      dsn: a str representing Postgres DSN string.
+    """
+    if self._connection and not self._connection.closed:
+      self.close()
+      self._connection = None
+    self._dsn = dsn
+
+  def get_connection(self):
+    """ Provides postgres connection. It can either return existing
+    working connection or establish new one.
+    Returns:
+      An instance of psycopg2 connection.
+    """
+    if not self._connection or self._connection.closed:
+      logger.info('Establishing new connection to Postgres server')
+      self._connection = psycopg2.connect(dsn=self._dsn)
+    return self._connection
+
+  def close(self):
+    """ Closes psycopg2 connection.
+    """
+    return self._connection.close()
+
+
+def init_table(pg_connection_wrapper):
+  from psycopg2 import IntegrityError  # Import psycopg2 lazily
+  # When multiple TQ servers are notified by ZK about new queue
+  # they sometimes get IntegrityError despite 'IF NOT EXISTS'
+  @retrying.retry(max_retries=5, retry_on_exception=IntegrityError)
+  def ensure_tables_created():
+    pg_connection = pg_connection_wrapper.get_connection()
+    with pg_connection:
+      with pg_connection.cursor() as pg_cursor:
+        pg_cursor.execute(
+          'CREATE TABLE IF NOT EXISTS "{table}" ('
+          '  email varchar(500) NOT NULL,'
+          '  password varchar(500) NOT NULL,'
+          '  date_creation timestamp NOT NULL,'
+          '  date_change timestamp NOT NULL,'
+          '  date_last_login timestamp NOT NULL,'
+          '  applications varchar(30)[],'
+          '  appdrop_rem_token varchar(500),'
+          '  appdrop_rem_token_exp bigint,'
+          '  visit_cnt integer,'
+          '  cookie varchar(4096),'
+          '  cookie_ip varchar(15),'
+          '  cookie_exp integer,'
+          '  cksum integer,'
+          '  enabled boolean,'
+          '  type varchar(9),'
+          '  is_cloud_admin boolean,'
+          '  capabilities varchar(255),'
+          '  PRIMARY KEY (email)'
+          ');'
+            .format(table=table_name)
+        )
```

Comment:
We might want to create a schema for the ua server so it has its own namespace, e.g.

```CREATE SCHEMA IF NOT EXISTS  "appscale_ua";```

though we would then need to check if the schema name can collide with other schemas (taskqueue)